tests/bluetooth/bsim/host/gatt/settings/test_scripts/run_gatt_settings.sh

```diff
@@ -26,10 +26,11 @@ function Execute() {
 cd ${BSIM_OUT_PATH}/bin
 
 # Remove the files used by the custom SETTINGS backend
-echo "remove settings files ${simulation_id}_*.log"
-rm ${simulation_id}_*.log || true
+TO_DELETE="${simulation_id}_server.log ${simulation_id}_client.log"
+echo "remove settings files ${TO_DELETE}"
+rm ${TO_DELETE} || true
```

Comment:
Is this to delete existing `bt_settings` stored? Or what is the SETTINGS in this context?

Reply:
yes, exactly what @aescolar says. this is the bsim equivalent of the `nrfjprog --recover` cmd.

Reply:
no, I have a `SETTINGS` backend that I stole from the mesh bsim tests. It just writes ascii to a standard file. 
This line is just so we start the test session with nothing stored.

Reply:
Ah, then it is as I thought. 

We don't have any guidelines on this AFAIK, but to me babblesim tests should be independent, i.e. you should be able to just run the test, without having to prepare (or in this case) delete possibly stored information first (to avoid issues like what we are seeing now). If the test is run in another other script or in another way, this needs to be fixed there too. 

Could the test itself not handle this? e.g. just clear settings at `init` if it requires that to run in the first place?

Reply:
The problem is due to the way bsim works, I can't join and disconnect devices from a running sim. So multiple devices need to access this settings file during the run.

This could maybe be moved to a shared-mem area, but I was pressed for time for a bugfix and this was the fastest way of getting a test upstream that verified the bugfix.

Reply:
I'm fine with fixing the test in any reasonable way. This PR is just a patch that fixes an intermittent failure in regression (either CI or locally) which is currently causing trouble. So if @jori-nordic  or you @Thalley want a nicer fix you are more than welcome to submit another PR. But if you think it will take more than a couple of hours to get there, I would think it will be much nicer for everybody that we merge this patch as a provisional fix.

Reply:
Yes I want to merge this too.

Reply:
Agreed. My comments were also mostly in the case that it was just a matter or doing `settings_delete()` or something from the code, rather than this. 

> The problem is due to the way bsim works, I can't join and disconnect devices from a running sim. So multiple devices need to access this settings file during the run.

Would it be possible for the last device in the simulator to delete the stored values just before `PASS`ing?

Reply:
yes, problem is if the last device asserts or panics. Doing it from the runner seems (or seemed) more robust

Reply:
> yes, problem is if the last device asserts or panics. Doing it from the runner seems (or seemed) more robust

Good point. Arguably both the first and last device should attempt cleanups then :D 
Because if you run the test outside of this script, then you still have the same issue. Fixing the issue in this script is not fixing the problem at the root